### DISCUSSION TITLE
After adding copypass, the scene display will change

### DISCUSSION THREAD

**Discussion by 53356454** [pmndrs/postprocessing#625] [Q&A]
import * as THREE from "three";
import { OrbitControls } from "three/examples/jsm/controls/OrbitControls.js";
import {
  EffectComposer,
  RenderPass,
  CopyPass,
} from "postprocessing";

const renderer = new THREE.WebGLRenderer({
  alpha: true,
  antialias: true,
});
renderer.setSize(window.innerWidth, window.innerHeight);

const container = document.getElementById("app");
// @ts-ignore
container.append(renderer.domElement);

const scene = new THREE.Scene();
const camera = new THREE.PerspectiveCamera(
  60,
  window.innerWidth / window.innerHeight,
  0.1,
  1000
);

const composer = new EffectComposer(renderer);
let renderPass = new RenderPass(scene, camera)
composer.addPass(renderPass);
const copyPass = new CopyPass();
composer.addPass(copyPass);

setInterval(() => {
  copyPass.enabled = !copyPass.enabled;

  renderPass.renderToScreen = !copyPass.enabled
  copyPass.renderToScreen = copyPass.enabled
}, 2000 )

const orbit = new OrbitControls(camera, renderer.domElement);
camera.position.set(10, 15, -22);
orbit.update();

const material = new THREE.MeshStandardMaterial({
  roughness: 0.95,
  color: new THREE.Color("#2aad2a"),
});
const planeMesh = new THREE.Mesh(new THREE.PlaneGeometry(2, 2), material);
planeMesh.rotateX(-Math.PI / 2);
scene.add(planeMesh);

const light = new THREE.AmbientLight(0xffffff);
scene.add(light);

const size = 10;
const divisions = 10;
const gridHelper = new THREE.GridHelper(size, divisions);
scene.add(gridHelper);

function animate(time) {
  requestAnimationFrame(animate);
  composer.render(time);
}

animate();

window.addEventListener("resize", function () {
  camera.aspect = window.innerWidth / window.innerHeight;
  camera.updateProjectionMatrix();
  renderer.setSize(window.innerWidth, window.innerHeight);
});

when 
copyPass.enabled = true
![1](https://github.com/pmndrs/postprocessing/assets/46946597/b4952e78-cf1d-46ac-b111-38a6fe58e27e)

copyPass.enabled = false
![2](https://github.com/pmndrs/postprocessing/assets/46946597/1b97de15-3770-4fe0-8ded-11ee5d2ddcb2)

when copyPass.enabled = true, How to set it to maintain its original state？Thanks

**Comment by vanruesc** [comment]
The colors in the two screenshots look the same to me. Are you asking about antialiasing?

**Reply by 53356454** [reply]
yes, about antialiasing. How to maintain the same antialiasing when adding copyPass？Thanks

**Reply by vanruesc** [reply]
MSAA can be enabled like this:

```js
// Using the constructor option:
const composer = new EffectComposer(renderer, {
  multisampling: 4
});

// Or using the property:
composer.multisampling = Math.min(16, renderer.capabilities.maxSamples:);
```

You can also use `SMAAEffect` or `FXAAEffect` for alternative image-based antialiasing.
